narwhal/consensus/src/utils.rs

```diff
@@ -62,21 +62,27 @@ pub fn order_dag(
     state: &ConsensusState,
 ) -> Vec<Certificate> {
     debug!("Processing sub-dag of {:?}", leader);
+    assert!(leader.round() > 0);
+    let gc_round = leader.round().saturating_sub(gc_depth);
+
     let mut ordered = Vec::new();
     let mut already_ordered = HashSet::new();
 
     let mut buffer = vec![leader];
     while let Some(x) = buffer.pop() {
         debug!("Sequencing {:?}", x);
         ordered.push(x.clone());
+        if x.round() == gc_round + 1 {
+            continue;
+        }
```

Comment:
I see the rationale here, but I believe this part is already covered/handled by the code bellow where we check the parents. I want to make sure at this point that we do only absolutely necessary additions.

Reply:
I can remove this block for now, but I don't like using `retain()` since it seems to scan over the whole map. I will include this block in another change to keep all certificates above gc round, if we get to that.